library/sha1.c

```diff
@@ -406,29 +439,30 @@ int mbedtls_sha1_self_test( int verbose )
         if( verbose != 0 )
             mbedtls_printf( "  SHA-1 test #%d: ", i + 1 );
 
-        mbedtls_sha1_starts( &ctx );
+        if( mbedtls_sha1_starts_ext( &ctx ) != 0 )
+            goto fail;
 
         if( i == 2 )
         {
             memset( buf, 'a', buflen = 1000 );
 
             for( j = 0; j < 1000; j++ )
-                mbedtls_sha1_update( &ctx, buf, buflen );
+            {
+                if( mbedtls_sha1_update_ext( &ctx, buf, buflen ) != 0 )
+                    goto fail;
+            }
         }
         else
-            mbedtls_sha1_update( &ctx, sha1_test_buf[i],
-                               sha1_test_buflen[i] );
+        {
+            if( mbedtls_sha1_update_ext( &ctx, sha1_test_buf[i],
+                                         sha1_test_buflen[i] ) != 0 )
+                goto fail;
+        }
 
-        mbedtls_sha1_finish( &ctx, sha1sum );
+        mbedtls_sha1_finish_ext( &ctx, sha1sum );
 
```

Comment:
This self-test is unusual compared to the other self-tests as the latter just call the main `mbedtls_XXX` function instead of the various stages.